app/View/Components/Submenu.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\View\Components;
+
+use Illuminate\View\Component;
+
+class Submenu extends Component
+{
+    /**
+     * The submenu title.
+     *
+     * @var string
+     */
+    public $title;
+
+    /**
+     * The submenu menu.
+     *
+     * @var string
+     */
+    public $menu;
+
+    /**
+     * The submenu device_id.
+     *
+     * @var string
+     */
+    public $device_id;
+
+    /**
+     * The submenu current_tab.
+     *
+     * @var string
+     */
+    public $current_tab;
+
+    /**
+     * The submenu selected.
+     *
+     * @var string
+     */
+    public $selected;
+
+    /**
+     * Create a new component instance.
+     *
+     * @return void
+     */
+    public function __construct($title, $menu, $deviceId, $currentTab, $selected)
+    {
+        $this->title = $title;
+        $this->menu = $menu;
+        $this->device_id = $deviceId;
+        $this->current_tab = $currentTab;
+        $this->selected = $selected;
```

Comment:
Looks like you changed selected from a string to an array, but didn't fully convert the code.
Perhaps renaming it to $tabs or $items might be a good idea.

Reply:
Hmm no, selected is a string with the current url To be able match currently selected tab.